src/librbd/internal.cc

```diff
@@ -72,6 +73,19 @@ namespace librbd {
 
 namespace {
 
+struct C_CreateV2Image : public Context {
```

Comment:
Never mind -- I see that you are using to clean up the temporary ImageCtx. You actually don't need the temporary ImageCtx since you've templated on it -- you can just pass the IoCtx, image id, and image name.